tests/dragonfly/replication_test.py

```diff
@@ -2096,3 +2103,43 @@ async def test_user_acl_replication(df_local_factory):
     await c_master.execute_command("ACL SETUSER tmp +replconf")
     await check_all_replicas_finished([c_replica], c_master, 5)
     assert 2 == await c_replica.execute_command("DBSIZE")
+
+
+@pytest.mark.asyncio
+async def test_replica_reconnect(df_local_factory):
+    # Test that a replica that disconnects from a master will not automatically replicate it after connection reestablishment, if it changed repl-id.
```

Comment:
This looks like a long line, dont we have a limit ?


    """
Test replica does not connect to master if master restarted   
    step1: create master and replica
    step2: stop master and start again with the same port
    step3: check replica is not replicating the restarted master
    step4: issue new replicaof command 
    step5: check replica replicates master
    """